src/Upgrade/Application/Dto/IntegratorResponseDto.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Upgrade\Application\Dto;
+
+class IntegratorResponseDto
+{
+    /**
+     * @var string
+     */
+    protected const MESSAGES_KEY = 'message-list';
+
+    /**
+     * @var string
+     */
+    protected const WARNINGS_KEY = 'warning-list';
+
+    /**
+     * @var array<mixed>
+     */
+    protected array $data = [];
+
+    /**
+     * @param array<mixed> $data
+     */
+    public function __construct(array $data = [])
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @return array<string>|null
+     */
+    public function getMessages(): ?array
+    {
+        if (!is_array($this->data[static::MESSAGES_KEY])) {
+            return null;
```

Comment:
actually, it is not necessary to return null here. You can return an empty array instead and have a single return type.

Reply:
Thanks ++